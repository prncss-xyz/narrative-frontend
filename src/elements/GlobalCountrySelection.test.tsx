import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { useRef } from "react";
import { act } from "react-dom/test-utils";
import {
  CountrySelectorContext,
  GlobalCountrySelection,
  useGlobalCountyList,
} from "./GlobalCountrySelection";

export const countries = [
  {
    countryCode: "US",
    name: "United States",
    storedData: [
      { datasetId: 1, recordCount: 500 },
      { datasetId: 2, recordCount: 0 },
      { datasetId: 3, recordCount: 2500 },
      { datasetId: 4, recordCount: 750 },
      { datasetId: 5, recordCount: 50 },
      { datasetId: 6, recordCount: 1000 },
    ],
  },
  {
    countryCode: "GB",
    name: "United Kingdom",
    storedData: [
      { datasetId: 1, recordCount: 0 },
      { datasetId: 2, recordCount: 1250 },
      { datasetId: 3, recordCount: 2000 },
      { datasetId: 4, recordCount: 1500 },
      { datasetId: 5, recordCount: 500 },
      { datasetId: 6, recordCount: 500 },
    ],
  },
];

function Container(spy: (activeCountryCodes: string[]) => void) {
  const activeCountryCodes = useGlobalCountyList(countries);
  // spy(activeCountryCodes);
  return (
    <CountrySelectorContext>
      <GlobalCountrySelection count={3} countries={countries} />;
    </CountrySelectorContext>
  );
}

describe.todo("GlobalCountrySelection", async () => {
  const { container } = render(
    <CountrySelectorContext>
      <GlobalCountrySelection count={3} countries={countries} />;
    </CountrySelectorContext>
  );
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should not be invisible", () => {
    expect(() => {
      screen.getByText("United States");
    }).toThrowError();
  });
  await userEvent.click(screen.getByText("Showing"));
  await screen.findByText("United States");
  // const s1 = JSON.stringify(screen.getByText("United States"));
  await userEvent.click(screen.getByText("United States"));
});
