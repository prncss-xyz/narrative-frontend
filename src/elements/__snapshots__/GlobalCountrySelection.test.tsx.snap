// Vitest Snapshot v1

exports[`GlobalCountrySelection > should match snapshot 1`] = `
<div>
  <div
    class="css-5uqfag-Box-Overlay edgsz5f2"
    height="100vh"
    width="100vw"
  />
  <div
    class="css-1q8bug2-Box-Flex-Overlay edgsz5f1"
    height="100vh"
    width="100vw"
  >
    <div
      class="css-18ya61d-Box-Flex edgsz5f1"
      height="100%"
      width="100%"
    >
      <div
        class="css-180d022-Box-Flex edgsz5f1"
        color="black"
      >
        <div
          class="css-13o8qag-Box edgsz5f2"
        >
          <div
            class="css-1x3sxzl-Box edgsz5f2"
          >
            <h3
              class="css-3dw4s3-Box edgsz5f2"
              color="gray1"
              font-size="2"
              font-weight="normal"
              text-decoration="underline"
            >
              Included countries
            </h3>
            <div
              class="css-vq3ac2-Box-Flex edgsz5f1"
            >
              <div>
                <button
                  class="css-3sr1pc-Box-RoundedButton edgsz5f2"
                >
                  United Kingdom
                </button>
              </div>
              <div>
                <button
                  class="css-arwt-Box-RoundedButton edgsz5f2"
                  color="gray1"
                >
                  United States
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    class="css-17pgxs5-Clickable"
  >
    <div
      class="css-1oaq5sg-Box-Flex edgsz5f1"
      color="gray1"
    >
      <div
        class="css-1x3sxzl-Box edgsz5f2"
      >
        Showing
      </div>
      <div
        class="css-t4uyl-Box edgsz5f2"
        color="black"
        font-weight="bold"
      >
         
        3
         
      </div>
      <div
        class="css-1x3sxzl-Box edgsz5f2"
      >
        results from
      </div>
      <div
        class="css-t4uyl-Box edgsz5f2"
        color="black"
        font-weight="bold"
      >
         
        United Kingdom
         
      </div>
    </div>
  </button>
  ;
</div>
`;
